### PR TITLE
Add Validation For Display Name Missing By Credentials Parameter

### PR DESCRIPTION
**Pull request by tomneeman151293**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/36317

## Description
Added special treatment for credentials parameter for display name not configured

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by tomneeman151293** [comment]
Added unit test to indicate bug.
Test failure:
https://app.circleci.com/pipelines/github/demisto/demisto-sdk/9633/workflows/2af32998-dfc7-4f8c-ae7e-bf4c5dc004f0/jobs/43242

File: demisto_sdk/commands/generate_docs/tests/generate_docs_test.py [hunk 1]
```diff
@@ -583,6 +586,27 @@ def test_generate_integration_doc(self):
...
+            - Test that credentials parameter name shown in README is using display password field.
+    """
+        readme = os.path.join(os.path.dirname(TEST_INTEGRATION_2_PATH), 'README.md')
+        # Generate doc
+        generate_integration_doc(TEST_INTEGRATION_2_PATH)
+        readme_data = open(readme).read()
+        assert readme_data == open(
+            os.path.join(os.path.dirname(TEST_INTEGRATION_2_PATH), 'README.md')).read()
+        assert '| None | The API key to use for the connection. | False |' not in readme_data
+        assert '| API Token | The API key to use for the connection. | False |'
```
Comment (dantavori) [inline demisto_sdk/commands/generate_docs/tests/generate_docs_test.py:608]: complete this assertion

File: demisto_sdk/commands/generate_docs/generate_integration_doc.py [hunk 2]
```diff
@@ -421,3 +420,27 @@ def command_permissions_filter(permission):
...
+        (None): Adds the data to 'access_data'.
+    """
+    display_name = credentials_conf.get('display')
+    display_password_name = credentials_conf.get('displaypassword', 'Password')
+    if display_name:
+        access_data.append(
+            {'Parameter': display_name,
+             'Description': string_escape_md(credentials_conf.get('additionalinfo', '')),
+             'Required': credentials_conf.get('required', '')})
+    if display_password_name:
```
Comment (dantavori) [inline demisto_sdk/commands/generate_docs/generate_integration_doc.py:442]: this is always true

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,7 @@
 # Changelog
 * Fixed the **generate-docs** to handle integration authentication parameter.
 * Added a validation to ensure that description and README do not contain the word 'Demisto'.
+* Fixed an issue where **generate-docs** command was displaying 'None' when credentials parameter display field configured was not configured
```
Comment (dantavori) [inline CHANGELOG.md:4]:
```suggestion
* Fixed an issue where **generate-docs** command was displaying 'None' when credentials parameter display field configured was not configured.
```

File: demisto_sdk/commands/generate_docs/generate_integration_doc.py (hunk 2 above)
Comment (dantavori) [reply demisto_sdk/commands/generate_docs/generate_integration_doc.py:442]: can we have both of the fields? make sure we don't append twice

**Review by dantavori — CHANGES_REQUESTED** [review]

File: demisto_sdk/commands/generate_docs/generate_integration_doc.py (hunk 2 above)
Comment (tomneeman151293) [reply demisto_sdk/commands/generate_docs/generate_integration_doc.py:442]: @dantavori 
Removed the if, anyway if both fields exist it should have 2 lines in the Parameter configuration, see this
https://github.com/demisto/etc/issues/37184

**Comment by tomneeman151293** [comment]
according to @yaakovi , the validate-files failure for the IF100 can be ignored for now (a fix for it will soon be made)

**Review by dantavori — APPROVED** [review]

**Pull request merged**